client/components/date-range/header.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent } from 'react';
+import { noop } from 'lodash';
+import { localize, LocalizeProps } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import SharedProps from './shared-props';
+
+interface Props {
+	onApplyClick: () => void;
```

Comment:
🤔 Technically, I believe the following is the most accurate:

```
	onApplyClick: React.MouseEventHandler< 'button' >;
```

That might make sense for places where we're interested in the `event.target` on the handler, but I don't believe that's the case here. Additionally, it relies on some internals of `Button` which is implemented in such a way that makes it difficult to reason about in terms of types.

Given that with a button we're usually interested in knowing that the button has been, I think we're good to stay with `() => void` as the handler type.

Reply:
Sounds good, I'll keep it as it is. It should be a relatively painless change if we ever need the more accurate version, in any case.